components/AddLink/AddLinkButton.tsx

```diff
@@ -1,19 +1,21 @@
-import { ReactNode } from "react";
 import classNames from "classnames";
 import styles from "./AddLinkButton.module.css";
 import ModalContainer from "../Modal/ModalContainer/ModalContainer";
 import useModal from "@/hooks/useModal";
+import AddLinkModalContent from "../Modal/AddLinkModalContent/AddLinkModalContent";
+import isValidURL from "@/utils/isValidURL";
 
 interface Props {
   inputValue: string;
-  children?: ReactNode;
+  folderListData: UserFolders[] | undefined;
 }
 
-function AddLinkButton({ inputValue, children }: Props) {
+function AddLinkButton({ inputValue, folderListData }: Props) {
```

Comment:
혹은, 꼭 필수로 있어야하는 prop인데 undefined인 상태를 고려하고싶다면
optional prop으로 만들지않고 빈 배열로 초기화 하는 경우도 있어요.

```suggestion
function AddLinkButton({ inputValue, folderListData = [] }: Props) {
```